File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> dict;
        vector<vector<string>> anagrams;

        for (const string& str : strs) {
            string sortedStr = sortStr(str);
            dict[sortedStr].push_back(str);
        }

        for (auto& pair : dict) {
            anagrams.push_back(pair.second);
        }

        return anagrams;
    }

private:
    string sortStr(const string& str) {
        string sorted = str;
        sort(sorted.begin(), sorted.end());
        return sorted;
    }
};

File: 49-group-anagrams/group-anagrams.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        Dictionary<string, List<string>> map = new Dictionary<string, List<string>>();

        for(int i = 0;i < strs.Length;i++)
        {
            string str = strs[i];
            string sorted = new string(str.OrderBy(c => c).ToArray());
            if(map.ContainsKey(sorted))
            {
                map[sorted].Add(str);
            }
            else
            {
                map[sorted] = new List<string>{str};
            }
        }

        IList<IList<string>> res = new List<IList<string>>();
        foreach(var pair in map)
        {
            res.Add(pair.Value);
        }

        return res;
    }
}